plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -434,6 +456,9 @@ get_device_locations (guint16 id)
 	case 0x15D3:
 	case 0x15DA:
 	case 0x15C0:
+	case 0x15E7:
+	case 0x15EA:
+	case 0x15EF:
 	case 0:
```

Comment:
Does anyone remember by any chance why 0 is here? (*embarrassed icon*)

Reply:
If I was gonna guess I'd say it's one of these:
1. it's for CRB/SDV
2. safe mode (let them use force)

Reply:
Rereading the comments on the original PR, it seems like we did it to be future-proof for devices.
https://github.com/hughsie/fwupd/pull/201#pullrequestreview-58681545

Assuming that the comparison of vendor and model IDs are enough to be safe.

(safe-mode isn't relevant. On safe-mode no device is recognized by FW as there is no FW running... And for host we must compare the swap-lane table to be sure it's OK and also the kernel driver must be updated anyway for it to work so it's fine to update fwupd too for this.)

The point is that now the 0 must return an empty array. Oops.

Reply:
@superm1 Does my reasoning here make sense? If so, I'd like to fix it before 1.0.6. (Should be quick and simple, and thus safe enough even w/o testing it, I think)

Reply:
(Ah, yes, this is what it returns at the end, line 608. If `id == 0` it returns `UNKNOWN_DEVICE`. OK, preparing a PR.)

Reply:
Okay